### PR TITLE
[v2k] Add Health Check and Graceful Shutdown

### PR DESCRIPTION
**Pull request by eduardogmisiuk**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature. It adds the ServiceStateManager module from the SDK to add health checking and graceful shutdown capabilities.

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR?** (Link to an issue, e.g. #99999)
https://github.com/dojot/dojot/issues/1848
https://github.com/dojot/dojot/issues/1846

* **Other information**:

### CODE REVIEW

File: connector/mqtt/vernemq/v2k-bridge/app/MQTTClient.js [hunk 1]
```diff
@@ -93,39 +108,40 @@ class MQTTClient {
...
    *
-   * @callback MQTTClient~onDisconnect
+   * @function onDisconnect
+   * @private
    */
   onDisconnect() {
-    this.logger.info(`Client ${this.mqttOptions.clientId} disconnected, reconnecting...`);
+    this.logger.warn(`Client ${this.mqttOptions.clientId} disconnected, reconnecting...`);
     this.isConnected = false;
-    this.mqttc.reconnect();
```
Comment (rascaraficci) [inline connector/mqtt/vernemq/v2k-bridge/app/MQTTClient.js:101]: When the MQTT broker restarts, does the v2k service reconnects automatically?

File: connector/mqtt/vernemq/v2k-bridge/Dockerfile [hunk 2]
```diff
@@ -48,3 +48,6 @@ COPY --from=base /opt/v2k_bridge /opt/v2k_bridge
 ENTRYPOINT ["/sbin/tini", "--"]
 
 CMD ["./bin/entryPoint.sh", "npm", "run", "app"]
+
+HEALTHCHECK --start-period=5s --interval=30s --timeout=5s --retries=2 \
```
Comment (rascaraficci) [inline connector/mqtt/vernemq/v2k-bridge/Dockerfile:52]: Wouldn't `retries=3` be better? 2 is not an usual value for this setting.

**Review by rascaraficci — COMMENTED** [review]
Good job! I left few comments.

File: connector/mqtt/vernemq/v2k-bridge/app/MQTTClient.js (hunk 1 above)
Comment (eduardogmisiuk) [reply connector/mqtt/vernemq/v2k-bridge/app/MQTTClient.js:101]: Yes, we do not need to explicitly call the reconnect function. There is the `reconnectPeriod` option in the (MQTT Client)[https://www.npmjs.com/package/mqtt#client] that enables the reconnection.

**Review by rascaraficci — APPROVED** [review]
It seems good to me!

**Review by jonaphael — APPROVED** [review]
Nice!

**Pull request merged**